rviz_default_plugins/src/rviz_default_plugins/displays/marker/markers/triangle_list_marker.cpp

```diff
@@ -261,6 +328,12 @@ bool TriangleListMarker::hasVertexColors(const MarkerBase::MarkerConstSharedPtr
   return new_message->colors.size() == new_message->points.size();
 }
 
+bool TriangleListMarker::hasTexture(const MarkerBase::MarkerConstSharedPtr new_message) const
+{
+  return !new_message->texture_map.empty() &&
+         new_message->uv_coordinates.size() == new_message->points.size();
```

Comment:
It doesn't appear that this method needs any information about the `TriangleListMarker` instance. I think it would be better to make this a free function. Then we could alternatively use it for other Marker types as well.

Reply:
For things like the arrow marker messages, the points will not equal the `uv_coordinates`. This is because the arrow has a pre-determined geometry. In other words, `points == coordinates` is particular to this message type.